crates/tags/src/tag.rs

```diff
@@ -1,60 +1,241 @@
 // SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
 // SPDX-License-Identifier: Apache-2.0
 
-use std::hash::Hash;
 use std::ops::{Deref, DerefMut};
 use std::str::FromStr;
 
-use drawbridge_http::http::{self, Error, Request, StatusCode};
+use drawbridge_http::http::Request;
 use drawbridge_http::{async_trait, FromRequest};
+use drawbridge_jose::jws::Jws;
+use drawbridge_type::Entry;
 
-use semver::{self, Version};
 use serde::{Deserialize, Serialize};
 
-#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
-pub struct Tag(Version);
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct TagName(String);
 
-impl Deref for Tag {
-    type Target = Version;
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct TagValue {
```

Comment:
This struct is redundant, instead we should add a `#[serde(flatten)]` `HashMap<String, json::Value>` field for the remaining fields within the `Entry` type.
See https://serde.rs/field-attrs.html#flatten

Reply:
I think this is beyond Nathaniel's vision for `Entry`. It's just to validate that the hash is there, then it's up to whichever object that uses it to handle the extra data. That's why I had `TagValue` store the original `Vec<u8>`. If anything, the flatten and `HashMap` should be in the `struct TagValue`.